### PR TITLE
perf(reorder): debouncing events + new algorithm

### LINKED ISSUE #7522: Reorder ion-items sometimes blocked together
Sometimes, when I am trying to perform a re-order, my `ion-items` get "grouped" together and I cannot drop at a certain index.  See the video here for the recreation.

### PR DESCRIPTION
**Pull request by manucorporat**
WIP 

DO NOT MERGE

BEFORE:
screen shot 2016-11-20 at 11 58 31

AFTER:
screen shot 2016-11-20 at 13 07 16

**Ionic Version**: 2.x

fixes #7522

### CODE REVIEW

File: src/components/item/item-reorder.ts [hunk 1]
```diff
@@ -248,32 +264,29 @@ export class ItemReorder {
...
+    // if (toIndex <= lastToIndex) {
+    //   for (var i = toIndex; i <= lastToIndex; i++) {
+    //     if (i !== fromIndex) {
+    //       (<any>children[i]).style[transform] = (i < fromIndex)
+    //         ? `translateY(${itemHeight}px)` : '';
+    //     }
+    //   }
+    // }
     let children = this._element.children;
+    console.log('from', fromIndex, 'to', toIndex);
```
Comment (leo6104) [inline src/components/item/item-reorder.ts]: console.log should be removed.

**Comment by brandyscarney** [comment]
Can we get the merge conflicts on this fixed and is it ready to merge or can it be?

**Comment by manucorporat** [comment]
This is going to be parte of V4

**Comment by FedericoAlvarez** [comment]
Are those changes currently merged? Because I'm having the `Item Reorder reports wrong indexes when dropped on original position` issue

**Pull request closed without merging**